app/apollo/test/cluster.fga.spec.js

```diff
@@ -0,0 +1,568 @@
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterByName
+  it('fgaUser01 has authorization to get cluster 1 by cluster name', async () => {
+    let response;
+    try {
+      const clusterId = testCluster1.cluster_id;
+      const clusterName = testCluster1.registration.name;
+      response = await clusterApi.byClusterName(fgaToken01, {
+        orgId: org01._id,
+        clusterName: clusterName,
+      });
+      const clusterByClusterName = response.data.data.clusterByName;
+      expect(clusterByClusterName.clusterId).to.equal(clusterId);
+      expect(clusterByClusterName.regState).to.equal('registering');  // record attr
+      expect(clusterByClusterName.status).to.equal('registered'); // created ~= updated
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterByName without authorization
+  it('fgaUser01 does NOT have authorization to get cluster noAuthCluster by cluster name', async () => {
+    let response;
+    try {
+      const clusterName = 'no-auth-cluster';
+      response = await clusterApi.byClusterName(fgaToken01, {
+        orgId: org01._id,
+        clusterName: clusterName,
+      });
+      expect(response.data.data.clusterByName).to.equal(null);
+      expect(response.data.errors[0].message).to.be.a('string');
+      expect(response.data.errors[0].message).to.contain('You are not allowed');
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clustersByOrgId
+  it('fgaUser01 has authorization to get all clusters by Org ID', async () => {
+    let response;
+    try {
+      response = await clusterApi.byOrgID(fgaToken01, {
+        orgId: org01._id,
+      });
+      const clustersByOrgId = response.data.data.clustersByOrgId;
+      expect(clustersByOrgId).to.be.an('array');
+      expect(clustersByOrgId).to.have.length(1);
+      expect(clustersByOrgId[0].resources).to.be.an('array');
+
+      // test skip and limit implementation
+      response = await clusterApi.byOrgID(fgaToken01, {
+        orgId: org01._id,
+        skip: 1, limit: 1,
+      });
+      expect(response.data.data.clustersByOrgId[0]).to.equal(undefined);
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clustersByOrgId without authorization
+  it('fgaUser02 does NOT have authorization to get all clusters by Org ID', async () => {
```

Comment:
fgauser02 technically _does_ have authorization to get all clusters, this will not return a 'You are not authorized' error.  Perhaps word it as `fgaUser02 gets zero authorized clusters when getting all clusters by Org ID`.

Reply:
Good change!

Reply:
good for all fga files

Reply:
resolved